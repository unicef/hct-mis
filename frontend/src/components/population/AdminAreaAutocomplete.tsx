import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import get from 'lodash/get';
import { InputAdornment } from '@material-ui/core';
import RoomRoundedIcon from '@material-ui/icons/RoomRounded';
import CircularProgress from '@material-ui/core/CircularProgress';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasLazyQuery,
} from '../../__generated__/graphql';
import { useBusinessArea } from '../../hooks/useBusinessArea';

const StyledAutocomplete = styled(Autocomplete)`
  width: ${(props) => (props.fullWidth ? '100%' : '232px')}
    .MuiFormControl-marginDense {
    margin-top: 4px;
  }
`;

export function AdminAreasAutocomplete({
  disabled,
  fullWidth,
  onFilterChange,
  name,
}: {
  disabled?;
  fullWidth?: boolean;
  onFilterChange;
  name: string;
}): React.ReactElement {
  const [open, setOpen] = useState(false);
  const [inputValue, onInputTextChange] = useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const businessArea = useBusinessArea();
  const [loadAdminAreas, { data, loading }] = useAllAdminAreasLazyQuery({
    variables: {
      first: 50,
      title: debouncedInputText,
      businessArea,
      level: 2,
    },
  });
  useEffect(() => {
    loadAdminAreas();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [debouncedInputText]);

  const onChangeMiddleware = (e, selectedValue): void => {
    onFilterChange((filters) => ({
      ...filters,
      [name]: selectedValue || undefined,
    }));
  };
  return (
    <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
      fullWidth={fullWidth}
      open={open}
      filterOptions={(options1) => options1}
      onChange={onChangeMiddleware}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={(e, reason) => {
        setOpen(false);
        if (reason === 'select-option') return;
        onInputTextChange('');
      }}
      getOptionSelected={(option, value1) => {
        return value1?.node?.id === option.node.id;
      }}
      getOptionLabel={(option) => {
        if (!option.node) {
          return '';
        }
        return `${option.node.title}`;
      }}
      disabled={disabled}
      options={get(data, 'allAdminAreas.edges', [])}
      loading={loading}
      renderInput={(params) => (
        <TextField
          {...params}
          label='Admin Level 2'
          variant='outlined'
          margin='dense'
          value={inputValue}
          onChange={(e) => onInputTextChange(e.target.value)}
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position='start'>
                <RoomRoundedIcon style={{ color: '#5f6368' }} />
              </InputAdornment>
            ),
            endAdornment: (
              <>
                {loading ? (
                  <CircularProgress color='inherit' size={20} />
                ) : null}
                {params.InputProps.endAdornment}
              </>
            ),
          }}
        />
      )}
    />
  );
}
