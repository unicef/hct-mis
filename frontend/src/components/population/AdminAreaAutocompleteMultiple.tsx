import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import get from 'lodash/get';
import { Box } from '@material-ui/core';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasQuery,
} from '../../__generated__/graphql';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { LoadingComponent } from '../LoadingComponent';
import { FieldLabel } from '../FieldLabel';

const StyledAutocomplete = styled(Autocomplete)`
  width: 232px;
  .MuiFormControl-marginDense {
    margin-top: 4px;
  }
`;

export function AdminAreaAutocompleteMultiple({
  value,
  onChange,
  disabled,
}: {
  value;
  onChange;
  disabled?;
}): React.ReactElement {
  const [open, setOpen] = React.useState(false);
  const [inputValue, setInputTextChange] = React.useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const [newValue, setNewValue] = useState([]);
  const businessArea = useBusinessArea();
  const { data, loading } = useAllAdminAreasQuery({
    variables: {
      first: 100,
      title: debouncedInputText,
      businessArea,
    },
  });
  useEffect(() => {
    setNewValue(value);
  }, [data, value]);
  useEffect(() => {
    setInputTextChange('');
  }, [value]);

  if (loading) return <LoadingComponent />;
  if (!data) return null;
  return (
    <Box display='flex' flexDirection='column'>
      <FieldLabel>Administrative Level 2</FieldLabel>
      <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
        open={open}
        multiple
        fullWidth
        filterOptions={(options1) => options1}
        onChange={onChange}
        value={newValue}
        onOpen={() => {
          setOpen(true);
        }}
        onClose={() => {
          setOpen(false);
        }}
        getOptionSelected={(option, value1) => {
          return value1?.node?.id === option.node.id;
        }}
        getOptionLabel={(option) => {
          if (!option.node) {
            return '';
          }
          return `${option.node.title}`;
        }}
        disabled={disabled}
        options={get(data, 'allAdminAreas.edges', [])}
        loading={loading}
        renderInput={(params) => {
          return (
            <TextField
              {...params}
              inputProps={{
                ...params.inputProps,
                value: inputValue,
              }}
              placeholder={
                newValue.length > 0 ? null : 'Administrative Level 2'
              }
              variant='outlined'
              margin='dense'
              value={inputValue}
              onChange={(e) => setInputTextChange(e.target.value)}
            />
          );
        }}
      />
    </Box>
  );
}
