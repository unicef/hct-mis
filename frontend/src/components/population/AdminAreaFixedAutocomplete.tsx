import { InputAdornment } from '@mui/material';
import CircularProgress from '@mui/material/CircularProgress';
import RoomRoundedIcon from '@mui/icons-material/RoomRounded';
import Autocomplete from '@mui/lab/Autocomplete';
import get from 'lodash/get';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasQuery,
} from '../../__generated__/graphql';

const StyledAutocomplete = styled(Autocomplete)`
  .MuiFormControl-marginDense {
    //margin-top: 4px;
  }
`;

export function AdminAreaFixedAutocomplete({
  value,
  onChange,
  disabled,
}: {
  value;
  onChange;
  disabled?;
}): React.ReactElement {
  const { t } = useTranslation();
  const [open, setOpen] = React.useState(false);
  const [inputValue, onInputTextChange] = React.useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const [newValue, setNewValue] = useState(null);
  const businessArea = useBusinessArea();
  const { data, loading } = useAllAdminAreasQuery({
    variables: {
      name: debouncedInputText,
      businessArea,
      first: 50,
      level: 2,
    },
  });

  useEffect(() => {
    if (data) {
      setNewValue(
        typeof value === 'string'
          ? data.allAdminAreas.edges.find((item) => item.node.name === value)
          : value,
      );
    } else {
      // setNewValue(value);
    }
    // onInputTextChange('');
  }, [data, value]);
  const onChangeMiddleware = (e, selectedValue, reason): void => {
    onInputTextChange(selectedValue?.node?.name);
    onChange(e, selectedValue, reason);
  };
  return (
    <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
      open={open}
      filterOptions={(options1) => options1}
      onChange={onChangeMiddleware}
      value={newValue}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={(e, reason) => {
        setOpen(false);
        if (value || reason === 'select-option') return;
        onInputTextChange(null);
      }}
      getOptionSelected={(option, selectedValue) => {
        return selectedValue?.node?.id === option.node.id;
      }}
      getOptionLabel={(option) => {
        if (!option.node) {
          return '';
        }
        return `${option.node.name}`;
      }}
      disabled={disabled}
      options={get(data, 'allAdminAreas.edges', [])}
      loading={loading}
      renderInput={(params) => (
        <TextField
          {...params}
          label={t('Administrative Level 2')}
          variant='outlined'
          margin='dense'
          value={inputValue}
          onChange={(e) => onInputTextChange(e.target.value)}
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position='start'>
                <RoomRoundedIcon style={{ color: '#5f6368' }} />
              </InputAdornment>
            ),
            endAdornment: (
              <>
                {loading ? (
                  <CircularProgress color='inherit' size={20} />
                ) : null}
                {params.InputProps.endAdornment}
              </>
            ),
          }}
        />
      )}
    />
  );
}
