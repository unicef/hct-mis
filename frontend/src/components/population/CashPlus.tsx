import React from 'react';
import styled from 'styled-components';
import { Typography, Paper, Grid } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  margin-bottom: ${({ theme }) => theme.spacing(6)}px;
`;

const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

export function CashPlus({ individual }): React.ReactElement {
  const { enrolledInNutritionProgramme, administrationOfRutf } = individual;
  return (
    <Overview>
      <Title>
        <Typography variant='h6'>Cash+</Typography>
      </Title>
      <Grid container spacing={6}>
        <Grid item xs={4}>
          <LabelizedField
            label='Enrolled in nutrition programme'
            value={enrolledInNutritionProgramme ? 'YES' : 'NO'}
          />
        </Grid>
        <Grid item xs={4}>
          <LabelizedField
            label='Administratiion of rutf'
            value={administrationOfRutf ? 'YES' : 'NO'}
          />
        </Grid>
      </Grid>
    </Overview>
  );
}
