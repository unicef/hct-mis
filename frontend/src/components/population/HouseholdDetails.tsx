import React from 'react';
import styled from 'styled-components';
import { Typography, Grid } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';
import { HouseholdNode } from '../../__generated__/graphql';
import { formatCurrency } from '../../utils/utils';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { MiśTheme } from '../../theme';
import { ContainerColumnWithBorder } from '../ContainerColumnWithBorder';

const Overview = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;
const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

const ContentLink = styled.a`
  font-family: ${({ theme }: { theme: MiśTheme }) =>
    theme.hctTypography.fontFamily};
  color: #253b46;
  font-size: 14px;
  line-height: 19px;
`;

interface HouseholdDetailsProps {
  houseHold: HouseholdNode;
}
export function HouseholdDetails({
  houseHold,
}: HouseholdDetailsProps): React.ReactElement {
  const businessArea = useBusinessArea();
  return (
    <ContainerColumnWithBorder>
      <Title>
        <Typography variant='h6'>Details</Typography>
      </Title>
      <Overview>
        <Grid container spacing={6}>
          <Grid item xs={4}>
            <LabelizedField label='Household Size'>
              <div>{houseHold.size}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Country'>
              <div>{houseHold.country || '-'}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Residence Status'>
              <div>{houseHold.residenceStatus}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Country of Origin'>
              <div>{houseHold.countryOrigin}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Head of Household'>
              <ContentLink
                href={`/${businessArea}/population/individuals/${houseHold.headOfHousehold.id}`}
              >
                {houseHold.headOfHousehold.fullName}
              </ContentLink>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Total Cash Received'>
              <div>{formatCurrency(houseHold.totalCashReceived)}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='PrOgRAmmE(S) ENROLLED'>
              <div>
                {houseHold.programs.edges.map((item) => (
                  <div>{item.node.name}</div>
                ))}
              </div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Administrative Level 1'>
              <div>{houseHold.address || '-'}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Administrative Level 2'>
              <div>{houseHold.adminArea?.title || '-'}</div>
            </LabelizedField>
          </Grid>
        </Grid>
      </Overview>
    </ContainerColumnWithBorder>
  );
}
