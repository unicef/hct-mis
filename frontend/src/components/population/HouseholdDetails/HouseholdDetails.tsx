import { Box, Grid, Paper, Typography } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { MiśTheme } from '../../../theme';
import { COLLECT_TYPES_MAPPING } from '../../../utils/constants';
import { choicesToDict, formatCurrencyWithSymbol } from '../../../utils/utils';
import {
  GrievancesChoiceDataQuery,
  HouseholdChoiceDataQuery,
  HouseholdNode,
} from '../../../__generated__/graphql';
import { ContentLink } from '../../core/ContentLink';
import { LabelizedField } from '../../core/LabelizedField';
import { Title } from '../../core/Title';
import {
  BigValue,
  BigValueContainer,
} from '../../rdi/details/RegistrationDetails/RegistrationDetails';
import { LinkedGrievancesModal } from '../LinkedGrievancesModal/LinkedGrievancesModal';

const Container = styled.div`
  display: flex;
  flex: 1;
  width: 100%;
  background-color: #fff;
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  flex-direction: column;
  align-items: center;
  border-color: #b1b1b5;
  border-bottom-width: 1px;
  border-bottom-style: solid;

  && > div {
    margin: 5px;
  }
`;

const Overview = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;
const OverviewPaper = styled(Paper)`
  margin: 20px 20px 0 20px;
  padding: 20px ${({ theme }) => theme.spacing(11)}px;
`;

const Label = styled.span`
  ${({ theme }: { theme: MiśTheme }) => theme.styledMixins.label}
`;

interface HouseholdDetailsProps {
  household: HouseholdNode;
  choicesData: HouseholdChoiceDataQuery;
  baseUrl: string;
  businessArea: string;
  grievancesChoices: GrievancesChoiceDataQuery;
}
export const HouseholdDetails = ({
  household,
  choicesData,
  baseUrl,
  businessArea,
  grievancesChoices,
}: HouseholdDetailsProps): React.ReactElement => {
  const { t } = useTranslation();
  const residenceChoicesDict = choicesToDict(
    choicesData.residenceStatusChoices,
  );
  return (
    <>
      <Container>
        <Title>
          <Typography variant='h6'>{t('Details')}</Typography>
        </Title>
        <Overview>
          <Grid container spacing={3}>
            <Grid item xs={3}>
              <LabelizedField label={t('Household Size')}>
                {household?.size}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Residence Status')}>
                {residenceChoicesDict[household?.residenceStatus]}
              </LabelizedField>
            </Grid>
            <Grid item xs={6}>
              <LabelizedField label={t('Head of Household')}>
                <ContentLink
                  href={`/${baseUrl}/population/individuals/${household?.headOfHousehold?.id}`}
                >
                  {household?.headOfHousehold?.fullName}
                </ContentLink>
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('FEMALE CHILD HEADED HOUSEHOLD')}>
                {household?.fchildHoh ? t('Yes') : t('No')}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('CHILD HEADED HOUSEHOLD')}>
                {household?.childHoh ? t('Yes') : t('No')}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Country')}>
                {household?.country}
              </LabelizedField>
            </Grid>

            <Grid item xs={3}>
              <LabelizedField label={t('Country of Origin')}>
                {household?.countryOrigin}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Address')}>
                {household?.address}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Village')}>
                {household?.village}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Zip code')}>
                {household?.zipCode}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Administrative Level 1')}>
                {household?.admin1?.name}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Administrative Level 2')}>
                {household?.admin2?.name}
              </LabelizedField>
            </Grid>
            <Grid item xs={6}>
              <LabelizedField label={t('Geolocation')}>
                {household?.geopoint
                  ? `${household?.geopoint?.coordinates[0]}, ${household?.geopoint?.coordinates[1]}`
                  : '-'}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('UNHCR CASE ID')}>
                {household?.unhcrId}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('LENGTH OF TIME SINCE ARRIVAL')}>
                {household?.flexFields?.months_displaced_h_f}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('NUMBER OF TIMES DISPLACED')}>
                {household?.flexFields?.number_times_displaced_h_f}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('IS THIS A RETURNEE HOUSEHOLD?')}>
                {household?.returnee ? t('Yes') : t('No')}
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              {household?.unicefId && (
                <LinkedGrievancesModal
                  household={household}
                  baseUrl={baseUrl}
                  businessArea={businessArea}
                  grievancesChoices={grievancesChoices}
                />
              )}
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('COLLECT TYPE')}>
                {COLLECT_TYPES_MAPPING[household?.collectIndividualData]}
              </LabelizedField>
            </Grid>
          </Grid>
        </Overview>
      </Container>
      <OverviewPaper>
        <Title>
          <Typography variant='h6'>{t('Benefits')}</Typography>
        </Title>
        <Grid container>
          <Grid item xs={4}>
            <Grid container>
              <Grid item xs={6}>
                <Label color='textSecondary'>
                  {t('PrOgRAmmE(S) ENROLLED')}
                </Label>
              </Grid>
              <Grid item xs={6}>
                <Label color='textSecondary'>{t('Cash received')}</Label>
              </Grid>
            </Grid>
            {household?.programsWithDeliveredQuantity?.length ? (
              household?.programsWithDeliveredQuantity?.map((item) => (
                <Box key={item.id} mb={2}>
                  <Grid container key={item.id}>
                    <Grid item xs={6}>
                      <ContentLink href={`/${baseUrl}/programs/${item.id}`}>
                        {item.name}
                      </ContentLink>
                    </Grid>
                    <Grid item xs={6}>
                      <Box display='flex' flexDirection='column'>
                        {item.quantity.map((qty) => (
                          <Box
                            key={`${item.id}-${qty.currency}-${qty.totalDeliveredQuantity}`}
                          >
                            {qty.currency === 'USD'
                              ? formatCurrencyWithSymbol(
                                  qty.totalDeliveredQuantity,
                                  qty.currency,
                                )
                              : `(${formatCurrencyWithSymbol(
                                  qty.totalDeliveredQuantity,
                                  qty.currency,
                                )})`}
                          </Box>
                        ))}
                      </Box>
                    </Grid>
                  </Grid>
                </Box>
              ))
            ) : (
              <Grid container>
                <Grid item xs={6}>
                  -
                </Grid>
                <Grid item xs={6}>
                  -
                </Grid>
              </Grid>
            )}
          </Grid>
          <Grid item xs={4}>
            <BigValueContainer>
              <LabelizedField label={t('Total Cash Received')}>
                <BigValue>
                  {formatCurrencyWithSymbol(
                    household?.totalCashReceivedUsd,
                    'USD',
                  )}
                </BigValue>
              </LabelizedField>
            </BigValueContainer>
          </Grid>
        </Grid>
      </OverviewPaper>
    </>
  );
};
