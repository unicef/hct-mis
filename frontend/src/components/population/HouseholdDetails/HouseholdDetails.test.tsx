import React from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { render } from '../../../testUtils/testUtils';
import { fakeHousehold } from '../../../../fixtures/population/fakeHousehold';
import { fakeHouseholdChoices } from '../../../../fixtures/population/fakeHouseholdChoices';
import { fakeApolloAllGrievances } from '../../../../fixtures/grievances/fakeApolloAllGrievances';
import { fakeGrievancesChoices } from '../../../../fixtures/grievances/fakeGrievancesChoices';
import { HouseholdDetails } from './HouseholdDetails';

describe('components/population/HouseholdDetails', () => {
  it('should render', () => {
    const { container } = render(
      <MockedProvider
        addTypename={false}
        mocks={fakeApolloAllGrievances}
      >
        <HouseholdDetails
          businessArea='afghanistan'
          household={fakeHousehold}
          choicesData={fakeHouseholdChoices}
          grievancesChoices={fakeGrievancesChoices}
        />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });
});