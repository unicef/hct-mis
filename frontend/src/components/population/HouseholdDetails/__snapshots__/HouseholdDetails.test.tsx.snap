// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/population/HouseholdDetails should render 1`] = `
<div>
  <div
    class="sc-ckVGcZ jaqOpI"
  >
    <div
      class="sc-bxivhb jsIVNl"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Details
      </h6>
    </div>
    <div
      class="sc-jKJlTe cydhVN"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Household Size
            </span>
            <div
              data-cy="label-Household Size"
            >
              4
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Residence Status
            </span>
            <div
              data-cy="label-Residence Status"
            >
              Displaced  |  Refugee / Asylum Seeker
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Head of Household
            </span>
            <div
              data-cy="label-Head of Household"
            >
              <a
                class="sc-bdVaJa iNmLrH"
                href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/population/individuals/SW5kaXZpZHVhbE5vZGU6ZjVmMzgwN2ItYTBjOS00Mjk0LTg0Y2QtNDhkMjllMjZhODU3"
              >
                Agata Kowalska
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              FEMALE CHILD HEADED HOUSEHOLD
            </span>
            <div
              data-cy="label-FEMALE CHILD HEADED HOUSEHOLD"
            >
              No
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              CHILD HEADED HOUSEHOLD
            </span>
            <div
              data-cy="label-CHILD HEADED HOUSEHOLD"
            >
              No
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Country
            </span>
            <div
              data-cy="label-Country"
            >
              Isle of Man
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Country of Origin
            </span>
            <div
              data-cy="label-Country of Origin"
            >
              San Marino
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Address
            </span>
            <div
              data-cy="label-Address"
            >
              938 Luna Cliffs Apt. 551
Jameschester, SC 24934
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Village
            </span>
            <div
              data-cy="label-Village"
            >
              -
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Zip code
            </span>
            <div
              data-cy="label-Zip code"
            >
              -
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Administrative Level 1
            </span>
            <div
              data-cy="label-Administrative Level 1"
            >
              Nangarhar
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Administrative Level 2
            </span>
            <div
              data-cy="label-Administrative Level 2"
            >
              Achin
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Geolocation
            </span>
            <div
              data-cy="label-Geolocation"
            >
              70.210209, 172.085021
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              UNHCR CASE ID
            </span>
            <div
              data-cy="label-UNHCR CASE ID"
            >
              -
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              LENGTH OF TIME SINCE ARRIVAL
            </span>
            <div
              data-cy="label-LENGTH OF TIME SINCE ARRIVAL"
            >
              12
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              NUMBER OF TIMES DISPLACED
            </span>
            <div
              data-cy="label-NUMBER OF TIMES DISPLACED"
            >
              -
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              IS THIS A RETURNEE HOUSEHOLD?
            </span>
            <div
              data-cy="label-IS THIS A RETURNEE HOUSEHOLD?"
            >
              No
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Linked Grievances
            </span>
            <div
              data-cy="label-Linked Grievances"
            >
              <span
                class="sc-htpNat cwQQYq"
                color="textSecondary"
              >
                <span>
                  -
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              COLLECT TYPE
            </span>
            <div
              data-cy="label-COLLECT TYPE"
            >
              -
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root sc-eNQAEJ kOdPfR MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="sc-bxivhb jsIVNl"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Benefits
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Cash received
          </span>
          <div
            data-cy="label-Cash received"
          >
            <div
              class="MuiBox-root MuiBox-root-1"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="MuiBox-root MuiBox-root-2"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-3"
                    >
                      USDÂ 3,251.00
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-4"
                    >
                      (BAMÂ 3,048.00)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-5"
                    >
                      (BNDÂ 2,959.00)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-6"
                    >
                      (ISKÂ 4,904)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-7"
                    >
                      (LSLÂ 2,505.00)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-8"
                    >
                      (TMTÂ 115.00)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-9"
                    >
                      (UAHÂ 484.00)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-10"
                    >
                      (UGXÂ 2,073)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-11"
                    >
                      (UZSÂ 895.00)
                    </div>
                    <div
                      class="MuiBox-root MuiBox-root-12"
                    >
                      (YERÂ 2,335)
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div
          class="sc-VigVT ZeWFY"
        >
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              Total Cash Received
            </span>
            <div
              data-cy="label-Total Cash Received"
            >
              <div
                class="sc-jTzLTM dhULVO"
              >
                USDÂ 24,697.00
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
